unlock-app/src/components/interface/MetadataTable.tsx

```diff
@@ -130,11 +131,12 @@ export const MetadataTable: React.FC<MetadataTableProps> = ({
   }
 
   const isKeyValid = (metadata: KeyMetadata) => {
-    if (!metadata?.expiration) return false
-    if (metadata?.expiration.toLowerCase() === 'never') return true
-    const now = new Date().getTime()
-    const expiration = new Date(metadata?.expiration).getTime()
-    return expiration > now
+    if (!metadata?.expiration) return // expiration not present
+
+    const expiration = expirationAsDate(metadata?.expiration)
```

Comment:
I don't think it is the rigth approach.
I assume `metadata?.expiration` is a number?

If the number is smaller than the timestamp for "now" then it is expired.
Otherwise it is not.

Reply:
this is because `metadata.expiration` is a string

```
interface KeyMetadata {
  // These 3 properties are always present -- they come down from the graph as
  // strings
  lockName: string
  expiration: string
  keyholderAddress: string
  // Can have any other arbitrary properies, as long as the values are strings.
  [key: string]: string
}
```

Reply:
`expiration` parsed as numbed to check as suggested